### PR TITLE
Add Level Controller and Door Collision

### PR DESCRIPTION
**Pull request by TomasMorton**

### CODE REVIEW

File: components/door.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Crafty.c("Door", {
+    init: function () {
+        this.requires('2D, DOM, Color, Collision');
+        this.attr({x: 0, y: 0, w: 20, h: 100});
+        this.color('#f85902');
```
Comment (TomasMorton) [inline components/door.js:5]: @baconeta you removed the `.color("orange")` from the door1 instantiation in level1,js, should I be removing the colour here also?

File: components/level_controller.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+Crafty.c("LevelController", {
```
Comment (TomasMorton) [inline components/level_controller.js:1]: Simple controller that loads between levels and stores the state for the current level index.

File: components/player.js [hunk 3]
```diff
@@ -53,7 +53,11 @@ Crafty.c("Player", {
         })
 
         this.onHit("Spike", (hitData) => {
-            this.resetLevel();
+            Crafty.trigger("ResetLevel");
+        });
+
+        this.onHit("Door", function () {
+            Crafty.trigger("NextLevel");
```
Comment (TomasMorton) [inline components/player.js:60]: Next level and reset are handled by the level controller through events now.

File: main.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
 var GAME_SCREEN_WIDTH = 1200;
 var GAME_SCREEN_HEIGHT = 800;
 
-Crafty.init(GAME_SCREEN_WIDTH,GAME_SCREEN_HEIGHT, document.getElementById('game'));
+Crafty.init(GAME_SCREEN_WIDTH, GAME_SCREEN_HEIGHT, document.getElementById('game'));
 
 Crafty.scene("Title");
+
+const levelController = Crafty.e("LevelController");
```
Comment (TomasMorton) [inline main.js:8]: Adding top level controllers (must have Persist as an inherited component)

**Review by StevenR152 — APPROVED** [review]
Looks great! 🥳

**Pull request merged**